runtime/cudaq/builder/kernel_builder.cpp

```diff
@@ -128,7 +128,16 @@ initializeBuilder(MLIRContext *context,
     int digit = rand() % 10;
     os << digit;
   }
+
+#if (defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER))
```

Comment:
I would suggest removing this block of defines and let the `FmtCore.h` file handle it (will need to add `-Wrestrict` there), then update this code to 
```cpp
kernelName += fmt::format("_{}", os.str());
```